util/template/expression_template.go

```diff
@@ -38,17 +38,24 @@ func expressionReplace(ctx context.Context, w io.Writer, expression string, env
 	var unmarshalledExpression string
 	err := json.Unmarshal(fmt.Appendf(nil, `"%s"`, expression), &unmarshalledExpression)
 	if err != nil && allowUnresolved {
-		log.WithError(err).Debug(ctx, "unresolved is allowed ")
+		log.WithError(err).Debug(ctx, "unresolved is allowed")
 		return fmt.Fprintf(w, "{{%s%s}}", kindExpression, expression)
 	}
 	if err != nil {
 		return 0, fmt.Errorf("failed to unmarshall JSON expression: %w", err)
 	}
 
+	if anyVarNotInEnv(unmarshalledExpression, []string{"item"}, env) && allowUnresolved {
```

Comment:
We should be able to combine this and retries into one call to anyVarNotInEnv, perhaps it could return the var that is not in env for the log line.